File: day_10/part1.cpp
#include <iostream>
using namespace std;

void fun(string input, int times);

int main()
{

    fun("1211",1);

    return 0;
}

void fun(string input, int times)
{
    int x = 0;
    string value;
    int count = 0;
    string newstring;

    for(int i = 0; i < input.size(); i++) {
        x = i;
        count = 0;
        do {
            x++;
            count++;
        }while(input[i] != input[x]);

        newstring += x += input[i];
    }
    
    cout << newstring << endl;
}

File: day_10/part1.rb
def chknum(n,x)
    return "#{x}.#{n}"
end

def und(input)

    array = input
    array_2 = Array.new
    count = 0
    i = 0
    checkn = 0

    array.each_with_index do |value,index|
        i = index;
        loop do
            checkn += 1
            i += 1
            index += 1
        count += 1
        break if value != array[i]
        end
        array_2 << chknum(value, count)
        break if checkn == array.length
        count = 0;
        i = 0;
    end

    return array_2
end

sux = [1,2,1,1]

2.times do
    sux = und(sux)
end


sux.each do |x|
    print x
end
